packages/web/components/complex/orders-history/index.tsx

```diff
@@ -156,54 +211,111 @@ export const OrderHistory = observer(() => {
+                  return (
+                    <tr key={row.id} className="h-[84px]">
+                      {row.getVisibleCells().map((cell) => (
+                        <td key={cell.id} className="!px-0 !text-left">
+                          {flexRender(
+                            cell.column.columnDef.cell,
+                            cell.getContext()
+                          )}
+                        </td>
+                      ))}
+                    </tr>
+                  );
+                })}
             </>
           )}
           {pendingOrders.length > 0 && (
             <>
-              <h6 className="pb-4 pt-8">Pending</h6>
-              {pendingOrders.map((row) => (
-                <tr key={row.id}>
-                  {row.getVisibleCells().map((cell) => (
-                    <td key={cell.id} className="!px-0 !text-left">
-                      {flexRender(
-                        cell.column.columnDef.cell,
-                        cell.getContext()
-                      )}
-                    </td>
-                  ))}
-                </tr>
-              ))}
+              <h6 className="h-[84px] pb-4 pt-8">Pending</h6>
+              {virtualRows
+                .filter(
+                  (row) =>
+                    row.index >=
+                      filledOrders.length + (filledOrders.length > 0 ? 1 : 0) &&
+                    row.index <
+                      filledOrders.length +
+                        (filledOrders.length > 0 ? 1 : 0) +
+                        pendingOrders.length
+                )
+                .map((virtualRow) => {
+                  const row =
+                    rows[
+                      virtualRow.index - (1 + filledOrders.length > 0 ? 1 : 0)
+                    ];
+                  return (
+                    <tr key={row.id} className="h-[84px]">
+                      {row.getVisibleCells().map((cell) => (
+                        <td key={cell.id} className="!px-0 !text-left">
+                          {flexRender(
+                            cell.column.columnDef.cell,
+                            cell.getContext()
+                          )}
+                        </td>
+                      ))}
+                    </tr>
+                  );
+                })}
             </>
           )}
           {pastOrders.length > 0 && (
             <>
-              <h6 className="pb-4 pt-8">Past</h6>
-              {pastOrders.map((row) => (
-                <tr key={row.id}>
-                  {row.getVisibleCells().map((cell) => (
-                    <td key={cell.id} className="!px-0 !text-left">
-                      {flexRender(
-                        cell.column.columnDef.cell,
-                        cell.getContext()
-                      )}
-                    </td>
-                  ))}
-                </tr>
-              ))}
+              <h6 className="h-[84px] pb-4 pt-8">Past</h6>
+              {virtualRows
```

Comment:
This was a little hard for me to follow. 

Suggestion:


```
const filledOrdersCount = filledOrders.length;
const pendingOrdersCount = pendingOrders.length;
const hasFilledOrders = filledOrdersCount > 0 ? 1 : 0;
const hasPendingOrders = pendingOrdersCount > 0 ? 1 : 0;
const additionalRows = hasFilledOrders + hasPendingOrders;

const startIndex = filledOrdersCount + pendingOrdersCount + additionalRows;
const endIndex = orders.length + additionalRows;

{virtualRows
  .filter((row) => row.index > startIndex && row.index <= endIndex)
  .map((virtualRow) => {
    const rowIndex = virtualRow.index - (1 + additionalRows);
    const row = rows[rowIndex];

    return (
      <tr key={row.id} className="h-[84px]">
        {row.getVisibleCells().map((cell) => (
          <td key={cell.id} className="!px-0 !text-left">
            {flexRender(cell.column.columnDef.cell, cell.getContext())}
          </td>
        ))}
      </tr>
    );
  })}
```

Reply:
I'd considered filtering the actual sub arrays in `useMemo` to make it cleaner

Reply:
It's a bit of a hacky solution so curious if there's a general cleaner approach

Reply:
Rewrote this to help with readability